backend/data/src/main/kotlin/io/tolgee/service/machineTranslation/MtService.kt

```diff
@@ -93,7 +94,18 @@ class MtService(
     publishBeforeEvent(project)
 
     checkTextLength(baseTranslationText)
-    val targetLanguageIds = targetLanguages.map { it.id }
+
+    // filter only translations that are not disabled
+    val targetLanguageIds = keyId?.let { kId ->
```

Comment:
I don't think this is a right place to do this. The purpose of this method is only to return the machine translations, not to handle disabled translations. Also, performance will suffer here, since you are retrieving key and all it's translations for every element in batch. 

The right place to do this would be `io.tolgee.service.translation.AutoTranslationService#softAutoTranslate` and `io.tolgee.service.translation.AutoTranslationService#autoTranslateSync`

This solution doesn't fix transaltion via translation memory.